docs/Makefile.am

```diff
@@ -0,0 +1,871 @@
+        shmem_put64_nbi.3 \
+        shmem_put8_nbi.3 \
+        shmem_putmem.3 \
+        shmem_putmem_nbi.3 \
+        shmem_quiet.3 \
+        shmem_realloc.3 \
+        shmem_set_cache_inv.3 \
+        shmem_set_cache_line_inv.3 \
+        shmem_set_lock.3 \
+        shmem_short_and_to_all.3 \
+        shmem_short_g.3 \
+        shmem_short_get.3 \
+        shmem_short_get_nbi.3 \
+        shmem_short_iget.3 \
+        shmem_short_iput.3 \
+        shmem_short_max_to_all.3 \
+        shmem_short_min_to_all.3 \
+        shmem_short_or_to_all.3 \
+        shmem_short_p.3 \
+        shmem_short_prod_to_all.3 \
+        shmem_short_put.3 \
+        shmem_short_put_nbi.3 \
+        shmem_short_sum_to_all.3 \
+        shmem_short_wait.3 \
+        shmem_short_wait_until.3 \
+        shmem_short_xor_to_all.3 \
+        shmem_swap.3 \
+        shmem_test_lock.3 \
+        shmem_udcflush.3 \
+        shmem_udcflush_line.3 \
+        shmem_wait.3 \
+        shmem_wait_until.3 \
+        shrealloc.3 \
+        start_pes.3
+
+MAN_OUTDIR = $(OUTDIR)/man
+
+OMPI_MAN1_RST = $(OMPI_MAN1:%.1=man-openmpi/man1/%.1.rst)
+OMPI_MAN1_BUILT = $(OMPI_MAN1:%.1=$(MAN_OUTDIR)/%.1)
+
+OMPI_MAN3_RST = $(OMPI_MAN3:%.3=man-openmpi/man3/%.3.rst)
+OMPI_MAN3_BUILT = $(OMPI_MAN3:%.3=$(MAN_OUTDIR)/%.3)
+
+OMPI_MAN5_RST = $(OMPI_MAN5:%.5=man-openmpi/man5/%.5.rst)
+OMPI_MAN5_BUILT = $(OMPI_MAN5:%.5=$(MAN_OUTDIR)/%.5)
+
+OSHMEM_MAN1_RST = $(OSHMEM_MAN1:%.1=man-oshmem/man1/%.1.rst)
+OSHMEM_MAN1_BUILT = $(OSHMEM_MAN1:%.1=$(MAN_OUTDIR)/%.1)
+
+OSHMEM_MAN3_RST = $(OSHMEM_MAN3:%.3=man-oshmem/man3/%.3.rst)
+OSHMEM_MAN3_BUILT = $(OSHMEM_MAN3:%.3=$(MAN_OUTDIR)/%.3)
+
+###########################################################################
+if OPAL_BUILD_DOCS
+
+include $(top_srcdir)/Makefile.ompi-rules
+
+# Have to not list these targets in EXTRA_DIST outside of the
+# OPAL_BUILD_DOCS conditional because "make dist" will fail due to
+# these missing targets (and therefore not run the "dist-hook" target
+# in the top-level Makefile, which prints a pretty message about why
+# "make dist" failed).
+#
+# We list the entire directory trees (html and man) to grab all
+# generated files in them.
+EXTRA_DIST += \
+        $(OUTDIR)/html \
+        $(OUTDIR)/man
+
+ALL_MAN_BUILT = \
+        $(OMPI_MAN1_BUILT) $(OMPI_MAN3_BUILT) $(OMPI_MAN5_BUILT) \
+        $(OSHMEM_MAN1_BUILT) $(OSHMEM_MAN_3_BUILT)
+$(ALL_MAN_BUILT): $(RST_SOURCE_FILES) $(IMAGE_SOURCE_FILES)
+$(ALL_MAN_BUILT): $(TEXT_SOURCE_FILES) $(SPHINX_CONFIG)
+
+# List both commands (HTML and man) in a single rule because they
+# really need to be run in serial.  Specifically, if they were two
+# different rules and someone ran "make -j", then both of them could
+# be writing to $(OUTDIR)/doctrees simultaneously, which would be Bad.
+# Use one of the man pages as a sentinel file to indicate whether all
+# the HTML docs and man pages have been built.
+$(ALL_MAN_BUILT):
+	$(SPHINX_BUILD) -M html "$(srcdir)" "$(OUTDIR)" $(SPHINX_OPTS)
```

Comment:
extra credit, we should really do some quiet thing here...

Reply:
What kind of quiet do you want:

1. `@` to not emit the command
1. Redirect stdout/stderr to `/dev/null`
    * I'm a little wary of this because running Sphinx can take 1-3 minutes, depending on the speed of your system/filesystem.
1. Use the Automake-quiet-like rules from Makefile.ompi-rules to hide both the command and the output
    * I'm wary of this for the same reason as the prior bullet

Reply:
Case in point -- on my Cisco i7 MacPro (admittedly with Cisco IT-installed anti-malware on it, which slows things down a little):

```
$ cd docs
$ /usr/bin/time make
...
       176.87 real       170.20 user         3.82 sys
```

Reply:
Yes, I think we should follow the Automake conventions and do the V= to set `@` or not.  While I'd like to have it also set `-q` to sphinx-build, I get what you're saying about time, so let's skip that for now.